### PR TITLE
Cb 4460 shared query history for the team

### PR DESCRIPTION
**Pull request by alexander-skoblikov**

### CODE REVIEW

File: server/bundles/io.cloudbeaver.service.admin/schema/service.admin.graphqls [hunk 1]
```diff
@@ -150,6 +157,8 @@ extend type Query {
 
     setUserAuthRole(userId: ID!, authRole: String): Boolean
 
+    setUserTeamRole(userId: ID!, teamId: ID!,teamRole: String): Boolean
+
```
Comment (Wroud) [inline server/bundles/io.cloudbeaver.service.admin/schema/service.admin.graphqls]:
```suggestion
    setUserTeamRole(userId: ID!, teamId: ID!, teamRole: String): Boolean

```

File: server/bundles/io.cloudbeaver.service.admin/schema/service.admin.graphqls [hunk 2]
```diff
@@ -53,6 +58,7 @@ type AdminTeamInfo {
     metaParameters: Object!
 
     grantedUsers: [ID!]!
+    grantedUsersInfo: [AdminUserTeamGrantInfo]!
```
Comment (Wroud) [inline server/bundles/io.cloudbeaver.service.admin/schema/service.admin.graphqls]:
```suggestion
    grantedUsersInfo: [AdminUserTeamGrantInfo!]!
```

File: webapp/packages/core-authentication/src/TeamsResource.ts [hunk 3]
```diff
@@ -85,9 +92,9 @@ export class TeamsResource extends CachedMapResource<string, TeamInfo, TeamResou
     return this.data;
   }
 
-  async loadGrantedUsers(teamId: string): Promise<string[]> {
+  async loadGrantedUsers(teamId: string): Promise<UserTeamGrantInfo[]> {
     const { team } = await this.graphQLService.sdk.getTeamGrantedUsers({ teamId });
-    return team[0].grantedUsers;
+    return team[0].grantedUsersInfo as UserTeamGrantInfo[];
```
Comment (Wroud) [inline webapp/packages/core-authentication/src/TeamsResource.ts]: ` as UserTeamGrantInfo[];` can be removed after schema fix

File: webapp/packages/plugin-authentication-administration/src/Administration/Users/Teams/GrantedUsers/useGrantedUsers.tsx [hunk 4]
```diff
@@ -49,10 +50,17 @@ export function useGrantedUsers(team: TeamInfo, mode: TeamFormMode): Readonly<St
...
-        this.state.grantedUsers.push(...subjectIds);
+        this.state.grantedUsers.push(...subjectIds.map(id => ({ userId: id, teamRole: null })));
+      },
+      assignTeamRole(subjectId: string, teamRole: string | null) {
+        const user = this.state.grantedUsers.find(user => user.userId === subjectId);
+
+        if (user) {
+          user.teamRole = teamRole;
+        }
       },
```
Comment (Wroud) [inline webapp/packages/plugin-authentication-administration/src/Administration/Users/Teams/GrantedUsers/useGrantedUsers.tsx:64]: seems like we can mutate object in some closer component that have access to it, right now we need to propagate a lot of logic through components

**Review by Wroud — CHANGES_REQUESTED** [review]

**Review by DenisSinelnikov — APPROVED** [review]

**Review by Wroud — APPROVED** [review]

**Pull request merged**